src/ophyd_async/core/signal.py

```diff
@@ -73,19 +75,47 @@ async def connect(
         backend: Optional[SignalBackend[T]] = None,
     ):
         if backend:
-            if self._initial_backend and backend is not self._initial_backend:
+            if self._backend and backend is not self._backend:
+                # test if the backend from a previous connection is the same
+                # reject even if there was no initial backend
                 raise ValueError(
                     "Backend at connection different from initialised one."
```

Comment:
I disagree, the comments explain more context

Reply:
But in this case I would argue the comment is misleading:

`"reject even if there was no initial backend"` - if there were no initial backend and it were the first connection then `self._initial_backend = self._backend = None` at this point and the condition in the if statement above would never be true?

I would also say `"test if the backend from a previous connection is the same"` right above a `ValueError("Backend at connection different from previous one.")` doesn't add anything in terms of readability.

Either way, the string in the error needs to changed from `"intitialised"` to `"previous"` because it will be checking against previous backends in general.